src/main/java/edu/cornell/kfs/concur/service/impl/ConcurAccountValidationServiceImpl.java

```diff
@@ -2,17 +2,218 @@
+        return validationResult;
+    }
+
+    public ValidationResult checkAccountingString(String chartOfAccountsCode, String accountNumber, String subAccountNumber, String objectCode, String subObjectCode, String projectCode) {
+        ValidationResult requiredFieldsValidationResult = checkRequiredAccountInfo(chartOfAccountsCode, accountNumber, objectCode);
+
+        if (requiredFieldsValidationResult.isNotValid()) {
+            return requiredFieldsValidationResult;
+        } else {
+            ValidationResult accountValidationResult = checkAccount(chartOfAccountsCode, accountNumber);
+            if (accountValidationResult.isNotValid()) {
+                return accountValidationResult;
+            } else {
+                ValidationResult objectCodeValidationResult = checkObjectCode(chartOfAccountsCode, objectCode);
+                ValidationResult subAccountValidationResult = checkSubAccount(chartOfAccountsCode, accountNumber, subAccountNumber);
+                ValidationResult subObjectCodeValidationResult = checkSubObjectCode(chartOfAccountsCode, accountNumber, objectCode, subObjectCode);
+                ValidationResult projectCodeValidationResult = checkProjectCode(projectCode);               
+                ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+
+                if (objectCodeValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(objectCodeValidationResult.getMessages());
+                }
+
+                if (subAccountValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(subAccountValidationResult.getMessages());
+                }
+
+                if (subObjectCodeValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(subObjectCodeValidationResult.getMessages());
+                }
+
+                if (projectCodeValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(projectCodeValidationResult.getMessages());
+                }
+
+                return validationResult;
+            }
+        }
+    }
+
+    public ValidationResult checkRequiredAccountInfo(String chartOfAccountsCode, String accountNumber, String objectCode) {
+        ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+        if (chartOfAccountsCode == null || chartOfAccountsCode.isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(ConcurConstants.AccountingStringValidationErrorMessages.ERROR_CHART_OF_ACCTS_REQUIRED);
+            return validationResult;
+        } else if (accountNumber == null || accountNumber.isEmpty()) {
```

Comment:
I'm wondering if these else if's are needed or if they should just be separate if blocks. As is, if the accountNumber is missing, that message won't be returned if the chart is missing too. Wouldn't it be helpful to have all the messages? Or are we limited to one message to return?